client/landing/gutenboarding/components/header/index.tsx

```diff
@@ -28,61 +25,21 @@ const Header: React.FunctionComponent = () => {
 	const { __, i18nLocale } = useI18n();
 	const currentStep = useCurrentStep();
 
-	const { domain, siteTitle } = useSelect( ( select ) => select( ONBOARD_STORE ).getState() );
+	const { siteTitle } = useSelect( ( select ) => select( ONBOARD_STORE ).getState() );
 
-	const [ domainSearch ] = useDebounce( siteTitle, 1000 /*ms*/ );
-
-	const recommendedDomain = useSelect(
-		( select ) => {
-			if ( ! domainSearch || domainSearch.length < 2 ) {
-				return;
-			}
-			return select( 'automattic/domains/suggestions' ).getDomainSuggestions( domainSearch, {
-				// Avoid `only_wordpressdotcom` — it seems to fail to find results sometimes
-				include_wordpressdotcom: false,
-				include_dotblogsubdomain: false,
-				quantity: 1, // this will give the recommended domain only
-				locale: i18nLocale,
-			} );
-		},
-		[ domainSearch ]
-	)?.[ 0 ];
-
-	const getDomainElementContent = () => {
-		// If no site title entered (user skips intent gathering)
-		// and no domain was selected, show "Pick a domain"
-		if ( ! siteTitle && ! domain ) {
-			return null;
-		}
-
-		if ( recommendedDomain && ! domain ) {
-			/* translators: domain name is available, eg: "yourname.com is available" */
-			return sprintf( __( '%s is available' ), recommendedDomain.domain_name );
-		}
-
-		return __( 'Choose a domain' );
-	};
-
-	/* eslint-disable wpcalypso/jsx-classname-namespace */
-	const domainElement = domain
-		? domain.domain_name
-		: getDomainElementContent() && (
-				<span className="gutenboarding__header-domain-picker-button-domain">
-					{ getDomainElementContent() }
-				</span>
-		  );
+	// steps (including modals) where we show Domains button
+	const showDomainsButton = [ 'DesignSelection', 'Style', 'Features', 'Plans' ].includes(
```

Comment:
Would be nice if these were constants instead of string. Looks like we don't have constants for step keys (or `StepNameType`) in `path.ts`. Nice-to-have for future PR.

Reply:
Could be good simple dev-onboarding task :-)